diff --git a/bazel/build_defs.bzl b/bazel/build_defs.bzl
index 5b35b277..556a9537 100644
--- a/bazel/build_defs.bzl
+++ b/bazel/build_defs.bzl
@@ -47,6 +47,10 @@ UPB_DEFAULT_COPTS = select({
         "-std=c99",
         "-pedantic",
         "-Werror=pedantic",
+        # That flag might be undefined on the CI, but is needed on the local
+        # MacOS build.
+        "-Wno-error=gnu-offsetof-extensions",
+        "-Wno-error=c2x-extensions",
         "-Wall",
         "-Wstrict-prototypes",
         # GCC (at least) emits spurious warnings for this that cannot be fixed
